packages/javascript/bh-shared-ui/src/components/AnalyzeNowConfiguration/AnalyzeNowConfiguration.tsx

```diff
@@ -0,0 +1,90 @@
+// Copyright 2025 Specter Ops, Inc.
+//
+// Licensed under the Apache License, Version 2.0
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+import { Button } from '@bloodhoundenterprise/doodleui';
+import { RequestOptions } from 'js-client-library';
+import { useState } from 'react';
+import { useMutation, useQuery, useQueryClient } from 'react-query';
+import { useNotifications } from '../../providers';
+import { apiClient } from '../../utils';
+import AnalyzeNowConfirmDialog from './AnalyzeNowConfirmDialog';
+
+const requestAnalysis = (options?: RequestOptions) => {
+    return apiClient.requestAnalysis(options).then((res) => res.data);
+};
+
+const useRequestAnalysis = () => {
+    const queryClient = useQueryClient();
+    return useMutation(requestAnalysis, {
+        onSuccess: () => {
+            queryClient.invalidateQueries('datapipe-status');
+        },
+    });
+};
+
+type AnalyzeNowProps = {
+    description: string;
+    note?: string;
+};
+
+const AnalyzeNowConfiguration: React.FC<AnalyzeNowProps> = ({ description, note }) => {
+    const [isOpenDialog, setIsOpenDialog] = useState(false);
+    const requestAnalysis = useRequestAnalysis();
+
+    const { addNotification } = useNotifications();
+
+    const showDialog = () => setIsOpenDialog(true);
+    const hideDialog = () => setIsOpenDialog(false);
+
+    const { data, isLoading, isError } = useQuery(
+        'datapipe-status',
+        ({ signal }) => apiClient.getDatapipeStatus({ signal }).then((res) => res.data?.data.status),
+        { refetchInterval: 2000 }
+    );
+    const buttonDisabled = isLoading || isError || data !== 'idle';
+
+    const handleConfirm = () => {
+        requestAnalysis.mutate(undefined, {
+            onError: () => {
+                hideDialog();
+                addNotification('There was an error requesting analysis.');
+            },
+            onSuccess: () => {
+                hideDialog();
+                addNotification('Analysis requested successfully.');
+            },
+        });
+    };
```

Comment:
And with all those changes, your mutation becomes real clean:

```suggestion
    const handleConfirm = () => {
        requestAnalysis()
    };
```

Technically, you could just include it directly as a component prop and delete this callback alltogether.

```tsx
<AnalyzeNowConfirmDialog open={isOpenDialog} onCancel={hideDialog} onConfirm={requestAnalysis} />
```